medusa/server/web/home/add_shows.py

```diff
@@ -167,6 +167,11 @@ def massAddTable(self, rootDir=None):
                     if not (indexer_id and show_name):
                         (indexer_id, show_name, indexer) = cur_provider.retrieveShowMetadata(cur_path)
 
+                        # unicode encoded special character show names, should be encoded to byte, as the nfo can have a
+                        # show title with unicode encoded special chars. Like naruto shippuuden on tvmaze.
+                        if show_name:
+                            show_name = show_name.encode('utf-8')
+
```

Comment:
After looking into it another time. I don't think it's in the wrong place. As i'm reading unicode here from file. You can't fully control what your are parsing. I think the application should be ready to handle it.

On a sidenote, I do agree that we should look into other issues, where we often log or parse unicode in mako. As for ex mako only supports bytes.